internal/kafka/internal/workers/kafka_mgrs/promotion/promotion_kafkas_mgr_test.go

```diff
@@ -0,0 +1,325 @@
+		kafkaConfig  config.KafkaConfig
+	}
+	type KafkaService_ListKafkasToBePromotedExpect struct {
+		calls int
+	}
+	type KafkaService_UpdateExpect struct {
+		promotionStatus dbapi.KafkaPromotionStatus
+		calls           int
+	}
+	type QuotaService_ReserveQuotaIfNotAlreadyReservedExpect struct {
+		calls int
+	}
+	type QuotaService_DeleteQuotaForBillingModelExpect struct {
+		calls int
+	}
+	type expect struct {
+		kafkaService_ListKafkasToBePromoted           KafkaService_ListKafkasToBePromotedExpect
+		kafkaService_Updates                          KafkaService_UpdateExpect
+		kafkaService_Update                           KafkaService_UpdateExpect
+		quotaService_ReserveQuotaIfNotAlreadyReserved QuotaService_ReserveQuotaIfNotAlreadyReservedExpect
+		quotaService_DeleteQuotaForBillingModel       QuotaService_DeleteQuotaForBillingModelExpect
+		wantDesiredBillingModel                       string
+		wantActualBillingModel                        string
+	}
+	tests := []struct {
+		name         string
+		fields       fields
+		wantErrCount int
+		expect       expect
+	}{
+		{
+			name:         "Test nothing to promote",
+			wantErrCount: 0,
+			fields: fields{
+				kafkaService: &services.KafkaServiceMock{
+					ListKafkasToBePromotedFunc: func() ([]*dbapi.KafkaRequest, *errors.ServiceError) {
+						return nil, nil
+					},
+				},
+				quotaService: &services.QuotaServiceMock{},
+			},
+			expect: expect{
+				kafkaService_ListKafkasToBePromoted: KafkaService_ListKafkasToBePromotedExpect{
+					calls: 1,
+				},
+			},
+		},
+		{
+			name:         "Test clean promotion",
+			wantErrCount: 0,
+			expect: expect{
+				wantDesiredBillingModel: "standard",
+				wantActualBillingModel:  "standard",
+				kafkaService_ListKafkasToBePromoted: KafkaService_ListKafkasToBePromotedExpect{
+					calls: 1,
+				},
+				kafkaService_Updates: KafkaService_UpdateExpect{
+					promotionStatus: "",
+					calls:           1,
+				},
+				quotaService_DeleteQuotaForBillingModel: QuotaService_DeleteQuotaForBillingModelExpect{
+					calls: 1,
+				},
+				quotaService_ReserveQuotaIfNotAlreadyReserved: QuotaService_ReserveQuotaIfNotAlreadyReservedExpect{
+					calls: 1,
+				},
+			},
+			fields: fields{
+				kafkaConfig: config.KafkaConfig{
+					Quota: &config.KafkaQuotaConfig{
+						Type:                         api.AMSQuotaType.String(),
+						AllowDeveloperInstance:       true,
+						MaxAllowedDeveloperInstances: 10,
+					},
+					SupportedInstanceTypes: &config.KafkaSupportedInstanceTypesConfig{
+						Configuration: config.SupportedKafkaInstanceTypesConfig{
+							SupportedKafkaInstanceTypes: []config.KafkaInstanceType{
+								{
+									Id:          "standard",
+									DisplayName: "standard",
+									Sizes:       nil,
+									SupportedBillingModels: []config.KafkaBillingModel{
+										{
+											ID:               "eval",
+											AMSResource:      "rhosak",
+											AMSProduct:       "RHOSAKEval",
```

Comment:
For the `AMSProduct` the constants defined in the `pkg/client/ocm` package can be used instead